tools/serverpod_cli/lib/src/migrations/generator.dart

```diff
@@ -118,6 +154,78 @@ class MigrationGenerator {
 
     return migrationVersion;
   }
+
+  Future<String?> repairMigration({
+    String? tag,
+    required bool force,
+    bool verbose = false,
+    required String runMode,
+  }) async {
+    var client = ConfigInfo(runMode).createServiceClient();
+    var versions = <String, String>{};
+
+    // Load the latest migration from all modules.
+    var modules = getMigrationModules();
+    var dstDefinitions = <DatabaseDefinition>[];
+    for (var module in modules) {
+      var version = await getLatestMigrationVersion(module: module);
```

Comment:
Perhaps not a huge issue, but waiting in a for loop means there is a blocking call on each iteration.

Could be written as 

```dart
final latestMigrations = modules.map((module) => getLatestMigrationVersion(module: module));

var versions = await Future.await(latestMigrations);
```

To load everything in parallel.

Combine with the above I would probably use a filter instead of the continue block too.

```dart
versions.where((version) => version != null);
```